src/mods/ModsManager.cpp

```diff
@@ -0,0 +1,256 @@
+            archive = dynamic_pointer_cast<Ship::Archive>(std::make_shared<Ship::FolderArchive>(path));
+        } else if (StringHelper::IEquals(extension, ".disabled")) {
+            // Skip disabled mods
+            continue;
+        } else {
+            SPDLOG_WARN("The file {} has an unrecognized extension ({}), ignoring it.", path, extension);
+            continue;
+        }
+
+        int nb_unknown_files = 0;
+        archive->Load();
+        ModMetadata metadata;
+        auto mods_file = archive->LoadFile("mods.toml");
+        if (mods_file != nullptr) {
+            metadata = ModMetadata::LoadFromTOML(std::string(mods_file->Buffer->data(), mods_file->Buffer->size()));
+            SPDLOG_INFO("Loaded mod: {}\n{}", metadata.name, metadata.ToString());
+        } else {
+            const std::string msg = "The archive at path " + path +
+                                    " is missing a mods.toml file. The Mod are likely incompatible.\n\n"
+                                    "Do you want to continue loading the mods?";
+            if (GameEngine::ShowYesNoBox("Missing mods.toml", msg.c_str()) == IDNO) {
+                exit(1);
+            }
+            metadata.name = std::filesystem::path(path).stem().string();
+            semver::parse("0.0.0", metadata.version);
+            SPDLOG_WARN("The mod at path {} is missing a mods.toml file. Using default metadata:\n{}", path, metadata.ToString());
+        }
+
+        AddModMetadata(metadata, archive);
+    }
+}
+
+void DetectCyclicDependencies() {
+    if (auto cyclicDeps = CheckCyclicDependencies(); cyclicDeps.has_value()) {
+        std::string msg = "Cyclic dependencies detected between the following mods:\n";
+        for (const auto& cycle : cyclicDeps.value()) {
+            msg += " - " + cycle + "\n";
+        }
+        msg += "\nPlease resolve these cyclic dependencies before continuing.\n";
+        GameEngine::ShowMessage("Cyclic Dependency Issues", msg.c_str());
+        exit(1);
+    }
+}
+
+void DetectOutdatedDependencies() {
+    bool exitDueToErrors = false;
+    std::string allDepIssues;
+
+    for (const auto& [meta, _] : Mods) {
+        if (auto toUpdate = ModToUpdateDependencies(meta); toUpdate.has_value()) {
+            exitDueToErrors = true;
+            allDepIssues += "The mod \"" + meta.name + "\" has the following missing or outdated dependencies:\n";
+            for (const auto& dep : toUpdate.value()) {
+                allDepIssues += " - " + dep + "\n";
+            }
+        }
+    }
+
+    if (exitDueToErrors) {
+        allDepIssues += "\nPlease resolve these dependency issues before continuing.\n";
+        GameEngine::ShowMessage("Dependency Issues", allDepIssues.c_str());
+        exit(1);
+    }
+}
+
+void PrintListOfMods() {
+    SPDLOG_INFO("List of loaded mods:");
+    for (const auto& [meta, _] : Mods) {
+        SPDLOG_INFO(" - {} v{}", meta.name, meta.version.to_string());
+    }
+}
+
+void InitModsSystem() {
```

Comment:
Can we move this function to the top of the file?

The idea is that the code reader should be able to understand the basic control flow. This being kinda the `main()` of this file per-say. This way you see the puzzle of how it's put together.